cmd/appliance/extract_logs.go

```diff
@@ -0,0 +1,174 @@
+package appliance
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/appgate/sdpctl/pkg/configuration"
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/factory"
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/appgate/journaldreader/journaldreader"
+)
+
+type logextractOpts struct {
+	Path string
+}
+
+func NewExtractLogsCmd(f *factory.Factory) *cobra.Command {
+	opts := logextractOpts{}
+	cmd := &cobra.Command{
+		Use:     "extract-logs",
+		Short:   docs.ApplianceExtractLogsDoc.Short,
+		Long:    docs.ApplianceExtractLogsDoc.Long,
+		Example: docs.ApplianceExtractLogsDoc.ExampleString(),
+		RunE: func(c *cobra.Command, args []string) error {
+			return logsExtractRun(args, &opts)
+		},
+		Annotations: map[string]string{
+			configuration.SkipAuthCheck: "true",
+		},
+	}
+	cmd.Flags().StringVar(&opts.Path, "path", ".", "Optional path to write to")
+	return cmd
+}
+
+func logsExtractRun(args []string, opts *logextractOpts) error {
+	var errs *multierror.Error
+	for i := 0; i < len(args); i++ {
+		log.Infof("Starting processing %s", args[i])
+		if err := processJournalFile(args[i], opts.Path); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+	return errs.ErrorOrNil()
+}
+
+func processJournalFile(file string, path string) error {
+	r, err := zip.OpenReader(file)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	var extractedFiles []string
+
+	// Iterate through the files in the archive,
+	// printing some of their contents.
+	for _, f := range r.File {
+		if strings.HasSuffix(f.Name, ".journal") {
+			log.Infof("Extracting journal file %s", f.Name)
+
+			rc, err := f.Open()
+			if err != nil {
+				return err
+			}
+
+			extracted, err := os.CreateTemp("", "")
+			if err != nil {
+				return err
+			}
+			extractedFiles = append(extractedFiles, extracted.Name())
+
+			_, err = io.Copy(extracted, rc)
+			if err != nil {
+				return err
+			}
+			rc.Close()
+			extracted.Close()
+		}
+	}
+
+	// Sort the extracted journald files
+	extractedFiles = journaldreader.SortJournalFiles(extractedFiles)
+
+	log.Infof("Extracting journal files complete. Processing...")
+
+	textlogs := make(map[string]*os.File)
+
+	// Parse the extracted files
+	for _, journalfile := range extractedFiles {
+		j := journaldreader.SdjournalReader{}
+
+		log.Infof("Processing %s...", journalfile)
+
+		err := j.Open(journalfile)
+		if err != nil {
+			continue
+		}
+
+		for {
+			entry, hasnext, err := j.Next()
+			if err != nil {
+				// We just move to the next log file in case of error
```

Comment:
Do we want a verbose mode where we log the errors here?

Reply:
Normally that happens at least once because the last log will be "incomplete" as it's still open for writing when being sent.